libp2p/P2PMessage.h

```diff
@@ -80,13 +87,20 @@ class P2PMessage : public dev::network::Message
     void encodeAMOPBuffer(std::string const& topic);
     virtual ssize_t decodeAMOPBuffer(std::shared_ptr<bytes> buffer, std::string& topic);
 
+    static void setCompressHandler(
+        std::shared_ptr<dev::compress::CompressInterface> _compressHandler)
+    {
+        m_compressHandler = _compressHandler;
+    }
+
 private:
     uint32_t m_length = 0;            ///< m_length = HEADER_LENGTH + length(m_buffer)
     PROTOCOL_ID m_protocolID = 0;     ///< message type, the first two bytes of information, when
                                       ///< greater than 0 is the ID of the request package.
     PACKET_TYPE m_packetType = 0;     ///< message sub type, the second two bytes of information
     uint32_t m_seq = 0;               ///< the message identify
     std::shared_ptr<bytes> m_buffer;  ///< message data
+    static std::shared_ptr<dev::compress::CompressInterface> m_compressHandler;
```

Comment:
why should this be static？ 
https://stackoverflow.com/questions/7026507/why-are-static-variables-considered-evil?rq=1
https://www.learncpp.com/cpp-tutorial/4-2a-why-global-variables-are-evil/
https://www.gamedev.net/forums/topic/647440-why-are-static-variables-bad/

Why are static variables considered evil?